main.py

```diff
@@ -876,6 +876,53 @@ def H_ii_indices(det_i: Determinant) -> Iterator[Two_electron_integral_index_pha
         for i, j in product(det_i.alpha, det_i.beta):
             yield (i, j, i, j), 1
 
+    @staticmethod
+    def category_C(idx, psi_i, psi_j, N_orb):
+        """
+        psi_i psi_j: Psi_det, lists of determinants
+        idx: i,j,k,l, index of integral <ij|kl>
+        For an integral i,j,k,l of category C, yield all dependent determinant pairs (I,J)
+        Possibilities are i = k < j < l: (1,2,1,3), i < k < j = l: (1,3,2,3), j < i = k < l: (2,1,2,3)
+        """
+        i, j, k, l = idx
+        (
+            spindet_a_occ_i,
+            spindet_b_occ_i,
+        ) = Hamiltonian_two_electrons_integral_driven.get_spindet_a_occ_spindet_b_occ(psi_i)
+
+        def do_category_C(i, j, k, l, psi_i, psi_j, spindet_occ_i, oppspindet_occ_i, spin, N_orb):
+            det_indices1 = (spindet_occ_i[i] & spindet_occ_i[j]) - spindet_occ_i[l]
+            det_indices2 = (oppspindet_occ_i[i] & spindet_occ_i[j]) - spindet_occ_i[l]
+            # Keep indices separate to include doubly counted determinants
+            for a in chain(det_indices1, det_indices2):
+                det = psi_i[a]
+                excited_spindet = Excitation(N_orb).apply_excitation(getattr(det, spin), [[j], [l]])
+                if spin == "alpha":
+                    excited_det = Determinant(excited_spindet, getattr(det, "beta"))
+                else:
+                    excited_det = Determinant(getattr(det, "alpha"), excited_spindet)
+                if excited_det in psi_j:
+                    I = psi_i.index(det)
```

Comment:
I guess `I == a` all the time no?

Reply:
Yes, but it doesn't get used outside of the if.

I meant to ask if there was a nice way of grabbing` I,J `while checking `if excited_det in psi_j`